polyply/src/map_to_molecule.py

```diff
@@ -121,7 +122,15 @@ def add_blocks(self, meta_molecule):
             if node + 1 in nx.get_node_attributes(new_mol, "resid").values():
                 continue
             block = force_field.blocks[resname]
-            new_mol.merge_molecule(block)
+            correspondance = new_mol.merge_molecule(block)
+
+            residue = nx.Graph()
+            for res_node in correspondance.values():
+                data = new_mol.nodes[res_node]
+                residue.add_node(res_node, **data)
```

Comment:
Use `nx.relabel_nodes` with `copy=True` with `correspondance` on `block` to generate `residue`? That way the intent is clearer.

Reply:
Yes and now these two things are actually not the same. I think because I'm  getting the attributes from new_mol node which change when to_molecule or merge is applied. In any case it doesn't work.